### PR TITLE
Soundcloud Command

### PR DESCRIPTION
**Pull request by metehus**
image
image
image
> **Note:** The playlist search returns random results. Thats an API issue

### CODE REVIEW

**Comment by metehus** [comment]
> LGTM

pls approve and dont just comment

File: src/apis/Soundcloud.js [hunk 1]
```diff
@@ -26,14 +26,30 @@ module.exports = class SoundcloudAPI extends APIWrapper {
     return this.request(`/tracks/${id}`, { secret_token: secret })
   }
 
+  getTrackById (id) {
+    return this.request(`/tracks/${id}`)
+  }
+
+  searchTrack (q) {
+    return this.request('/tracks/', { q })
```
Comment (davipatury) [inline src/apis/Soundcloud.js]:
```suggestion
    return this.request('/tracks', { q })
```

File: src/apis/Soundcloud.js [hunk 2]
```diff
@@ -26,14 +26,30 @@ module.exports = class SoundcloudAPI extends APIWrapper {
...
+  }
+
   getUser (id) {
     return this.request(`/users/${id}`)
   }
 
+  searchUser (q) {
+    return this.request('/users/', { q })
```
Comment (davipatury) [inline src/apis/Soundcloud.js]:
```suggestion
    return this.request('/users', { q })
```

File: src/apis/Soundcloud.js [hunk 3]
```diff
@@ -26,14 +26,30 @@ module.exports = class SoundcloudAPI extends APIWrapper {
...
+  }
+
   getPlaylist (id) {
     return this.request(`/playlists/${id}`)
   }
 
+  searchPlaylist (q) {
+    return this.request('/playlists/', { q })
```
Comment (davipatury) [inline src/apis/Soundcloud.js]:
```suggestion
    return this.request('/playlists', { q })
```

File: src/commands/misc/soundcloud/playlist.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      parentCommand: 'soundcloud',
+      embedColor: Constants.SOUNDCLOUD_COLOR,
+      embedLogoURL: 'https://i.imgur.com/s8e2oin.png'
+    })
+  }
+
+  async search (context, query) {
+    const results = await this.client.apis.soundcloud.searchPlaylist(query)
+    return results
+  }
```
Comment (davipatury) [inline src/commands/misc/soundcloud/playlist.js]:
```suggestion
  search (context, query) {
    return this.client.apis.soundcloud.searchPlaylist(query)
  }
```

File: src/commands/misc/soundcloud/track.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      parentCommand: 'soundcloud',
+      embedColor: Constants.SOUNDCLOUD_COLOR,
+      embedLogoURL: 'https://i.imgur.com/s8e2oin.png'
+    })
+  }
+
+  async search (context, query) {
+    const results = await this.client.apis.soundcloud.searchTrack(query)
+    return results
+  }
```
Comment (davipatury) [inline src/commands/misc/soundcloud/track.js]:
```suggestion
  search (context, query) {
    return this.client.apis.soundcloud.searchTrack(query)
  }
```

File: src/commands/misc/soundcloud/user.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+      parentCommand: 'soundcloud',
+      embedColor: Constants.SOUNDCLOUD_COLOR,
+      embedLogoURL: 'https://i.imgur.com/s8e2oin.png'
+    })
+  }
+
+  async search (context, query) {
+    const results = await this.client.apis.soundcloud.searchUser(query)
+    return results
+  }
```
Comment (davipatury) [inline src/commands/misc/soundcloud/user.js]:
```suggestion
  search (context, query) {
    return this.client.apis.soundcloud.searchUser(query)
  }
```

**Comment by metehus** [comment]
whats missing?

**Comment by pedrofracassi** [comment]
Solve the conflicts please? :)

**Comment by metehus** [comment]
which conflicts

**Comment by almeidx** [comment]
> which conflicts

https://github.com/SwitchbladeBot/switchblade/pull/969/conflicts

**Comment by pedrofracassi** [comment]
> > which conflicts
> 
> [#969 (conflicts)](https://github.com/SwitchbladeBot/switchblade/pull/969/conflicts)

@metehus

**Comment by metehus** [comment]
how do i fix it here

_This was sent using Github for Mi Band 4_

**Comment by pedrofracassi** [comment]
> how do i fix it here
> 
> _This was sent using Github for Mi Band 4_

https://github.com/SwitchbladeBot/switchblade/pull/969/conflicts

**Comment by metehus** [comment]
someone did some shit on the code and now it's fucked up, big changes needed

**Comment by pedrofracassi** [comment]
So... how's this going?

**Comment by metehus** [comment]
f

**Comment by Doges** [comment]
eu vou fazer essa porra denovo quer ver

**Comment by Doges** [comment]
ta tudo quebrado nessa porra

**Pull request closed without merging**